tools/integration_tests/util/creds_tests/creds.go

```diff
@@ -21,50 +21,75 @@ import (
 
 	"cloud.google.com/go/compute/metadata"
 	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
 	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
 )
 
 const NameOfServiceAccount = "creds-test-gcsfuse"
+const CredentialsBucketName = "gcsfuse-integration-tests-data"
 
-func setPermission(permission string, serviceAccount string) {
-	// Provide permission to the bucket.
-	setup.RunScriptForTestData("../util/creds_tests/testdata/provide_permission.sh", setup.TestBucket(), serviceAccount, permission)
-}
+var WhitelistedGcpProjects = []string{"gcs-fuse-test", "gcs-fuse-test-ml"}
 
 func RunTestsForKeyFileAndGoogleApplicationCredentialsEnvVarSet(testFlagSet [][]string, permission string, m *testing.M) (successCode int) {
+	log.Println("Running credentials tests...")
+
 	// Fetching project-id to get service account id.
 	id, err := metadata.ProjectID()
 	if err != nil {
 		log.Printf("Error in fetching project id: %v", err)
+		return 1
+	}
+	// return if active GCP project is not in whitelisted gcp projects
+	if !slices.Contains(WhitelistedGcpProjects, id) {
+		log.Printf("The active GCP project is not one of: %s. So the credentials test will not run.", strings.Join(WhitelistedGcpProjects, ", "))
+		return 0
 	}
 
 	// Service account id format is name@project-id.iam.gserviceaccount.com
 	serviceAccount := NameOfServiceAccount + "@" + id + ".iam.gserviceaccount.com"
 
-	// Create service account
-	setup.RunScriptForTestData("../util/creds_tests/testdata/create_service_account.sh", NameOfServiceAccount)
+	localKeyFilePath := path.Join(os.Getenv("HOME"), "creds.json")
+	gcsKeyFilePath := path.Join(CredentialsBucketName, id+".json")
 
-	key_file_path := path.Join(os.Getenv("HOME"), "creds.json")
-
-	// Create credentials
-	setup.RunScriptForTestData("../util/creds_tests/testdata/create_key_file.sh", key_file_path, serviceAccount)
+	// Download credentials
+	err = operations.DownloadGcsObject(gcsKeyFilePath, localKeyFilePath)
+	if err != nil {
+		log.Printf("Error while downloading key file %v", err)
+		return 1
+	}
 
 	// Provide permission to service account for testing.
-	setPermission(permission, serviceAccount)
-
-	// Revoke the permission and delete creds after testing.
-	defer setup.RunScriptForTestData("../util/creds_tests/testdata/revoke_permission_and_creds.sh", serviceAccount, key_file_path)
+	_, err = operations.ExecuteGsutilCommandf(fmt.Sprintf("iam ch serviceAccount:%s:%s gs://%s", serviceAccount, permission, setup.TestBucket()))
+	if err != nil {
+		log.Printf("Error while setting permissions to SA: %v", err)
+		return 1
+	}
+	// Waiting for 2 minutes as it usually takes within 2 minutes for policy
+	// changes to propagate: https://cloud.google.com/iam/docs/access-change-propagation
+	time.Sleep(120 * time.Second)
+
+	// Revoke the permission after testing.
+	defer func(format string) {
```

Comment:
Should we keep this thing in function for better readability and call that function defer function.